drivers/CpuUid.h

```diff
@@ -0,0 +1,106 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_CPUUID_H
+#define MBED_CPUUID_H
+
+#include "platform/platform.h"
+
+#if defined(DEVICE_CPUUID) || defined(DOXYGEN_ONLY)
+
+#include <string>
+#include <vector>
+
+namespace mbed {
+/** \addtogroup drivers */
+
+/** CPU UID reader class
+ *
+ * @note Synchronization level: Interrupt safe
+ * @ingroup drivers
+ */
+class CpuUid {
+public:
+    /** CPU UID array typedef
+     */
+    typedef std::vector<uint8_t> cpu_uid_array_t;
```

Comment:
same comment about vector as for string. should not be used in APIs or return types